trackpy/linking/partial.py

```diff
@@ -0,0 +1,155 @@
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+import six
+from six.moves import range, zip
+import itertools
+import warnings
+import logging
+
+import numpy as np
+
+from ..utils import guess_pos_columns, validate_tuple, pandas_sort
+from .linking import link_iter
+
+logger = logging.getLogger(__name__)
+
+
+def coords_from_df_partial(df, pos_columns, t_column, link_frame_nos):
+    for frame_no in link_frame_nos:
+        yield frame_no, df.loc[df[t_column] == frame_no, pos_columns].values
+
+
+def link_partial(f, search_range, link_range,
+                 pos_columns=None, t_column='frame', **kwargs):
+    """Link a DataFrame of coordinates into trajectories.
+
+    Parameters
+    ----------
+    f : DataFrame
+        The DataFrame must include any number of column(s) for position and a
+        column of frame numbers. By default, 'x' and 'y' are expected for
+        position, and 'frame' is expected for frame number. See below for
+        options to use custom column names.
+    search_range : float or tuple
+        the maximum distance features can move between frames,
+        optionally per dimension
+    link_range : tuple
+        Only in the range(start, stop) will be analyzed.
+    memory : integer, optional
+        the maximum number of frames during which a feature can vanish,
+        then reappear nearby, and be considered the same particle. 0 by default.
+    pos_columns : list of str, optional
+        Default is ['y', 'x'], or ['z', 'y', 'x'] when 'z' is present in f
+    t_column : str, optional
+        Default is 'frame'
+    predictor : function, optional
+        Improve performance by guessing where a particle will be in
+        the next frame.
+        For examples of how this works, see the "predict" module.
+    adaptive_stop : float, optional
+        If not None, when encountering an oversize subnet, retry by progressively
+        reducing search_range until the subnet is solvable. If search_range
+        becomes <= adaptive_stop, give up and raise a SubnetOversizeException.
+    adaptive_step : float, optional
+        Reduce search_range by multiplying it by this factor.
+    link_strategy : {'recursive', 'nonrecursive', 'numba', 'drop', 'auto'}
+        algorithm used to resolve subnetworks of nearby particles
+        'auto' uses numba if available
+        'drop' causes particles in subnetworks to go unlinked
+
+    Returns
+    -------
+    DataFrame with added column 'particle' containing trajectory labels.
+    The t_column (by default: 'frame') will be coerced to integer."""
```

Comment:
Should `"""` be on a new line to better separate docstring from code?

Reply:
Add something like: 

```
A large dataset can be divided into several patches (time intervals) and linked 
separately with this function, for e.g. parallel processing. The results can then 
be stitched back together with reconnect_traj_patch().

Notes
-----
The memory option cannot retain particles at the boundaries between 
patches. Likewise, if a predictor depends on past trajectories, it may 
work poorly at the start of each patch.
```